db/common/urbo_upsertmetadata.sql

```diff
@@ -0,0 +1,419 @@
+
+--------------------------------------------------------------------------------
+HOW TO USE:
+PgSQL:
+SELECT urbo_upsertmetadata(json);
+
+json Example:
+{
+  "schools.institute.position": {
+      "var_name" : "Posición del instituto"
+    },
+  "irrigation.humiditysensor": {
+      "table_name": "irrigation_humiditysensor",
+      "mandatory" : true
+    },
+  "dumps": {
+      "config": { "carto": { "account": "cedus-admin" } },
+      "nodata": false
+    }
+}
+Call Example:
+SELECT urbo_upsertmetadata('{"schools.institute.position":{"var_name":"Posición del instituto"},"irrigation.humiditysensor":{"table_name":"irrigation_humiditysensor","mandatory":true},"dumps":{"config":{"carto":{"account":"cedus-admin"}},"nodata":false}}'::json);
+
+----------------------------------------
+Category complete json definition
+
+{
+  "category_name": -- example: "dumps"
+  {
+    "id_scopes_in": array of strings, -- if not defined it is applied generally and for all the existing scopes.
+    "id_scopes_out": array of strings,
+    "id_category": character varying(255), -- NOT NULL
+    "category_name": character varying(255),
+    "nodata": boolean, -- DEFAULT false
+    "config": jsonb -- DEFAULT ''{}''::jsonb
+  }
+}
+
+----------------------------------------
+Entity complete json definition
+
+{
+  "category_name.entity_name": -- example: "irrigation.humiditysensor"
+  {
+    "id_scopes_in": array of strings, -- if not defined it is applied generally and for all the existing scopes.
+    "id_scopes_out": array of strings,
+    "id_entity": character varying(255), -- NOT NULL
+    "entity_name": character varying(255),
+    "id_category": character varying(255),
+    "table_name": character varying(255),
+    "mandatory": boolean, -- DEFAULT false
+    "editable": boolean -- DEFAULT false
+  }
+}
+
+----------------------------------------
+Variable complete json definition
+
+{
+  "category_name.entity_name.variable_name": -- example: "schools.institute.position"
+  {
+    "id_scopes_in": array of strings, -- if not defined it is applied generally and for all the existing scopes.
+    "id_scopes_out": array of strings,
+    "id_variable": character varying(255), -- NOT NULL
+    "id_entity": character varying(255), -- NOT NULL
+    "entity_field": character varying(255), -- NOT NULL
+    "var_name": character varying(255),
+    "var_units": character varying(255),
+    "var_thresholds": double precision[],
+    "var_agg": character varying[],
+    "var_reverse": boolean,
+    "config": jsonb,
+    "table_name": character varying(255),
+    "type": character varying(255), -- DEFAULT ''catalogue''::character varying
+    "mandatory": boolean, -- DEFAULT false
+    "editable": boolean -- DEFAULT false
+  }
+}
+--------------------------------------------------------------------------------
+*/
+
+DROP FUNCTION IF EXISTS urbo_upsertmetadata(jsonb);
+CREATE OR REPLACE FUNCTION urbo_upsertmetadata(
```

Comment:
I don't understand why this function performs that many conversions between `jsonb` and `json`.